#476 Import recipes with multiple instruction steps (for a custom Co…

MarcusWolschon opened the pull request:
- Implement a custom scraper for Cookidoo with handling of stite specific formatings
- Implement the ability to manually or automatically import multi-step receipts by using the "Header 1" markdown
- automatically assign ingredients to the first step that mentions them when importing a multi step receipt

vabene1111 commented:
i dont see how this respects what we have written in previous issues and in the discussion we had in discord ?

How can the user quickly and easily decide if he wants to split steps or not ?

What happens when i import from a page that gives me 25 steps? do i get 25 markdown headers ?

how is this represented in the frontend? i do not see any changes to the importer UI ?

MarcusWolschon commented:
You change it by choosing 'manual' after reviewing the preview and noticing that manual adjustment is needed and then editing the markdown before using the "import" button. 

It's currently only used for this one site that is known to have reasonable steps since the receipts there are not user supplied. (no json+ld offered for the user supplied receipts)

MarcusWolschon commented:
I lack the expertise and insight to extend the preview page. That could be done on the feature brance before considering to merge this into the development branch at a later point.
However GitHub didn't offer me to Create such a branch and then doing the PR to that one.

MarcusWolschon commented:
Improvements:

1. preview shows steps preformated
2. preview and preview-editing shows an text about the use of markdown
3. text of the header is used as the name of the step
4. localised default step names for English and German

vabene1111 commented:
i will review this after my vacation and decide how we proceed

MarcusWolschon commented:
@vabene1111 I added an environment variable ENABLE_IMPORT_STEPS similar to ENABLE_PDF_EXPORT, so the experimental feature has to be explicitly enabled by the admin.

vabene1111 commented:
closed for compliance reasons

MarcusWolschon commented:
Added unit test case for the "cookidoo" scraper,
improved the translations
and added the ability to work with complex receips that have grouped steps.
(generating not only level 1 but also level 2,... headers in the instructions)
